.github/workflows/docker-build-and-run.yml

```diff
@@ -0,0 +1,85 @@
+name: Build and Run Docker
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, edited]
+    branches:
+      - main
+      - "docker/**"
+jobs:
+  build-and-run-docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Generate input files
+        run: |
+          python -m Docker.generate_signal_docker_test
+
+      - name: Verify input files
+        run: |
+          for file in ivim_simulation.nii.gz ivim_simulation.bval ivim_simulation.bvec; do
+            if [ ! -f "$file" ]; then
+              echo "Error: $file not found"
+              exit 1
+            fi
+          done
+          echo "All input files generated successfully"
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build Docker image
+        run: |
+          docker build -t tf2.4_ivim-mri_codecollection -f Docker/Dockerfile .
+
+      - name: Save and compress Docker image 
+        run: |
+          docker save tf2.4_ivim-mri_codecollection | gzip > tf2.4_ivim-mri_codecollection.tar.gz
+
+      - name: Upload Docker image artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: docker-image
+          path: tf2.4_ivim-mri_codecollection.tar.gz
```

Comment:
I'm not seeing why, but the artifact is empty. I wonder if it's too big? If there's no way to save it perhaps it could be split or maybe we just forget about saving the file for now.

Reply:
I will check this and verify.

Reply:
@etpeterson, I think it is there, where are you checking 
![Screenshot From 2024-10-01 05-29-09](https://github.com/user-attachments/assets/3a8dfb57-483b-4e30-a681-86d1c996a86d)

Reply:
A whole docker image is not 198 bytes. If you download it and unzip it you'll find it empty. Also: 
<img width="1032" alt="image" src="https://github.com/user-attachments/assets/04691625-876c-4da4-a9fb-62fba13d03a7">

Reply:
@etpeterson, any suggested solution ?

Reply:
You could try a cleaning 
` - name: Cleanup Docker
     run: |
       docker system prune -a --force`

You could also try alpine linux base. I think it's smaller but might not work correctly.

If neither of those works then let's build and test but not push.